README.rst
Python Client for Cloud VM Migration API
========================================

|stable| |pypi| |versions|

`Cloud VM Migration API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-vm-migration.svg
   :target: https://pypi.org/project/google-cloud-vm-migration/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-vm-migration.svg
   :target: https://pypi.org/project/google-cloud-vm-migration/
.. _Cloud VM Migration API: https://cloud.google.com/migrate/compute-engine/docs
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/vmmigration/latest
.. _Product Documentation:  https://cloud.google.com/migrate/compute-engine/docs

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Cloud VM Migration API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Cloud VM Migration API.:  https://cloud.google.com/migrate/compute-engine/docs
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-vm-migration


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-vm-migration

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Cloud VM Migration API
   to see other available methods on the client.
-  Read the `Cloud VM Migration API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Cloud VM Migration API Product documentation:  https://cloud.google.com/migrate/compute-engine/docs
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

## [1.4.0](https://github.com/googleapis/python-vm-migration/compare/v1.3.1...v1.4.0) (2022-09-07)


### Features

* Bump gapic-generator-python version to 1.3.0 ([681cc63](https://github.com/googleapis/python-vm-migration/commit/681cc634ad5963b122384268c676d719e02b5c6e))

## [1.3.1](https://github.com/googleapis/python-vm-migration/compare/v1.3.0...v1.3.1) (2022-08-11)


### Bug Fixes

* **deps:** allow protobuf < 5.0.0 ([#69](https://github.com/googleapis/python-vm-migration/issues/69)) ([d4d1ba8](https://github.com/googleapis/python-vm-migration/commit/d4d1ba873f490e30a85efb8a2df8c0ca3edf8daa))
* **deps:** require proto-plus >= 1.22.0 ([d4d1ba8](https://github.com/googleapis/python-vm-migration/commit/d4d1ba873f490e30a85efb8a2df8c0ca3edf8daa))

## [1.3.0](https://github.com/googleapis/python-vm-migration/compare/v1.2.0...v1.3.0) (2022-07-19)


### Features

* add ApplianceVersion, AvailableUpdates, MigratingVmView, UpgradeApplianceRequest, UpgradeApplianceResponse, UpgradeStatus ([#64](https://github.com/googleapis/python-vm-migration/issues/64)) ([839fac4](https://github.com/googleapis/python-vm-migration/commit/839fac47189552905a80d8443df90cd8f97829fe))

## [1.2.0](https://github.com/googleapis/python-vm-migration/compare/v1.1.2...v1.2.0) (2022-07-16)


### Features

* add audience parameter ([61827e2](https://github.com/googleapis/python-vm-migration/commit/61827e246c7aae16537d00095737a47ccf537f90))


### Bug Fixes

* **deps:** require google-api-core>=1.32.0,>=2.8.0 ([#61](https://github.com/googleapis/python-vm-migration/issues/61)) ([2e7fb1e](https://github.com/googleapis/python-vm-migration/commit/2e7fb1ef0d7069cd22f64de464148940c8330ac2))
* require python 3.7+ ([#59](https://github.com/googleapis/python-vm-migration/issues/59)) ([af250ac](https://github.com/googleapis/python-vm-migration/commit/af250ac6e3307ce002b5c2cedd3878342e580c7e))

## [1.1.2](https://github.com/googleapis/python-vm-migration/compare/v1.1.1...v1.1.2) (2022-06-03)


### Bug Fixes

* **deps:** require protobuf <4.0.0dev ([#48](https://github.com/googleapis/python-vm-migration/issues/48)) ([c5c3155](https://github.com/googleapis/python-vm-migration/commit/c5c3155f62d5f46ac4f5071d68af0c448edcd93d))


### Documentation

* fix changelog header to consistent size ([#49](https://github.com/googleapis/python-vm-migration/issues/49)) ([1f5f92a](https://github.com/googleapis/python-vm-migration/commit/1f5f92ab6422e9b737a9f8e597501eb0cf17b798))

## [1.1.1](https://github.com/googleapis/python-vm-migration/compare/v1.1.0...v1.1.1) (2022-03-05)


### Bug Fixes

* **deps:** require google-api-core>=1.31.5, >=2.3.2 ([#24](https://github.com/googleapis/python-vm-migration/issues/24)) ([0ee06fd](https://github.com/googleapis/python-vm-migration/commit/0ee06fda92a66981f21b0fe546335362f4fc8d80))

## [1.1.0](https://github.com/googleapis/python-vm-migration/compare/v1.0.0...v1.1.0) (2022-02-11)


### Features

* add api key support ([#14](https://github.com/googleapis/python-vm-migration/issues/14)) ([bf6760c](https://github.com/googleapis/python-vm-migration/commit/bf6760ce5ead26b352a5a89e079fa2ca20c0c3c6))


### Bug Fixes

* resolve DuplicateCredentialArgs error when using credentials_file ([c53543e](https://github.com/googleapis/python-vm-migration/commit/c53543e159c2513089223fdc956860a051244c29))

## [1.0.0](https://github.com/googleapis/python-vm-migration/compare/v0.1.0...v1.0.0) (2022-01-24)


### Features

* bump release level to production/stable ([#4](https://github.com/googleapis/python-vm-migration/issues/4)) ([23b4a5e](https://github.com/googleapis/python-vm-migration/commit/23b4a5ef93452580a4587e3d95163fcf664ed39f))

## 0.1.0 (2021-11-18)


### Features

* generate v1 ([4041de0](https://www.github.com/googleapis/python-vm-migration/commit/4041de00804957fddba57f6e972c7ed1415354f9))


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2022 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

ALL_PYTHON = [
    "3.6",
    "3.7",
    "3.8",
    "3.9",
    "3.10",
]

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")

BLACK_VERSION = "black==22.3.0"
BLACK_PATHS = ["docs", "google", "tests", "samples", "noxfile.py", "setup.py"]
DEFAULT_PYTHON_VERSION = "3.10"

nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
    "blacken",
    "lint",
    "lint_setup_py",
]

@nox.session(python=ALL_PYTHON)
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/vmmigration_v1/',
        '--cov=tests/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=ALL_PYTHON)
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python=DEFAULT_PYTHON_VERSION)
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx==4.0.1", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint(session):
    """Run linters.

    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", BLACK_VERSION)
    session.run(
        "black",
        "--check",
        *BLACK_PATHS,
    )
    session.run("flake8", "google", "tests", "samples")


@nox.session(python=DEFAULT_PYTHON_VERSION)
def blacken(session):
    """Run black. Format code to uniform standard."""
    session.install(BLACK_VERSION)
    session.run(
        "black",
        *BLACK_PATHS,
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint_setup_py(session):
    """Verify that setup.py is valid (including RST check)."""
    session.install("docutils", "pygments")
    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
